### PR TITLE
Next15 support

### PR DESCRIPTION
**Pull request by thomasballinger**

### CODE REVIEW

**Comment by thomasballinger** [comment]
Iterating separately from Chadwick's branch, https://github.com/get-convex/convex-auth/pull/118

**Review by sshader — APPROVED** [review]
So we've gotten some testing with Next 15 and the alpha version with this code -- as long as we've also manually tested that Next 14 still works (after adding all the `await`s) I'm down to ship it

**Comment by thomasballinger** [comment]
Just manually tested, looks good with a Next.js 14 site.

File: docs/pages/authz/nextjs.mdx [hunk 1]
```diff
@@ -28,10 +26,10 @@ const isSignInPage = createRouteMatcher(["/signin"]);
 const isProtectedRoute = createRouteMatcher(["/product(.*)"]);
 
 export default convexAuthNextjsMiddleware((request, { convexAuth }) => {
```
Comment (edproton) [inline docs/pages/authz/nextjs.mdx:28]: Is lacking `async` here.

**Review by edproton — COMMENTED** [review]
Improve docs

**Pull request merged**